src/components/screens/vacation-requests/vacation-requests-screen.tsx

```diff
@@ -0,0 +1,287 @@
+      try {
+        const foundVacationRequestStatus = vacationRequestStatuses.find(
+          (vacationRequestStatus) => vacationRequestStatus.vacationRequestId === selectedRow
+        );
+        if (foundVacationRequestStatus?.id) {
+          await vacationRequestStatusApi.deleteVacationRequestStatus({
+            id: foundVacationRequestStatus.id,
+            statusId: foundVacationRequestStatus.id
+          });
+          const filteredVacationRequestStatuses = vacationRequestStatuses.filter(
+            (vacationRequestStatus) => vacationRequestStatus.id !== foundVacationRequestStatus.id
+          );
+          setVacationRequestStatuses(filteredVacationRequestStatuses);
+        }
+      } catch (error) {
+        setError(`${strings.vacationRequestError.deleteStatusError}, ${error}`);
+      }
+    }
+  };
+
+  /**
+   * Delete vacation requests
+   *
+   * @param selectedRowIds selected row ids
+   */
+  const deleteVacationRequests = async (selectedRowIds: GridRowId[]) => {
+    if (vacationRequests) {
+      let updatedVacationRequests: VacationRequest[] = [];
+      await Promise.all(
+        selectedRowIds.map(async (selectedRowId) => {
+          try {
+            await deleteVacationRequestStatus(selectedRowId);
+            await vacationRequestsApi.deleteVacationRequest({
+              id: selectedRowId as string
+            });
+            updatedVacationRequests = vacationRequests.filter(
+              (vacationRequest) => vacationRequest.id !== selectedRowId
+            );
+          } catch (error) {
+            setError(`${strings.vacationRequestError.deleteRequestError}, ${error}`);
+          }
+        })
+      );
+      setVacationRequests(updatedVacationRequests);
+    }
+  };
+
+  /**
+   * Create a vacation request status
+   *
+   * @param createdRequest created vacation request
+   */
+  const createVacationRequestStatus = async (createdRequest: VacationRequest) => {
+    if (!userProfile || !userProfile.id) return;
+
+    try {
+      if (createdRequest.id) {
+        const createdStatus = await vacationRequestStatusApi.createVacationRequestStatus({
+          id: createdRequest.id,
+          vacationRequestStatus: {
+            vacationRequestId: createdRequest.id,
+            status: VacationRequestStatuses.PENDING,
+            message: createdRequest.message,
+            createdAt: new Date(),
+            createdBy: userProfile.id,
+            updatedAt: new Date(),
+            updatedBy: userProfile.id
+          }
+        });
+
+        setVacationRequestStatuses([createdStatus, ...vacationRequestStatuses]);
+      }
+    } catch (error) {
+      setError(`${strings.vacationRequestError.createStatusError}, ${error}`);
+    }
+  };
+
+  /**
+   * Create a vacation request
+   *
+   * @param vacationData vacation data
+   */
+  const createVacationRequest = async (vacationData: VacationData) => {
+    if (!userProfile || !userProfile.id) return;
```

Comment:
You could use optional chaining here e.g. !userProfile?.id

Reply:
Used optional chaining:
`if (!userProfile?.id) return;`